### PR TITLE
Fix versions for 2.x on main

### PR DESCRIPTION
**Pull request by kotwanikunal**
### Description
- Fix version for `2.x` post version bump on `main`

### Related Issues
- N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 [![Security Vulnerabilities](https://img.shields.io/github/issues/opensearch-project/OpenSearch/security%20vulnerability?labelColor=red)](https://github.com/opensearch-project/OpenSearch/issues?q=is%3Aissue+is%3Aopen+label%3A"security%20vulnerability")
 [![Open Issues](https://img.shields.io/github/issues/opensearch-project/OpenSearch)](https://github.com/opensearch-project/OpenSearch/issues)
 [![Open Pull Requests](https://img.shields.io/github/issues-pr/opensearch-project/OpenSearch)](https://github.com/opensearch-project/OpenSearch/pulls)
-[![2.8 Open Issues](https://img.shields.io/github/issues/opensearch-project/OpenSearch/v2.8.0)](https://github.com/opensearch-project/OpenSearch/issues?q=is%3Aissue+is%3Aopen+label%3A"v2.8.0")
+[![2.9 Open Issues](https://img.shields.io/github/issues/opensearch-project/OpenSearch/v2.9.0)](https://github.com/opensearch-project/OpenSearch/issues?q=is%3Aissue+is%3Aopen+label%3A"v2.9.0")
```
Comment (andrross) [inline README.md:10]: Should this be 2.10? Is the intent here to show the open issues going into the next minor release?

Comment (kotwanikunal) [reply README.md:10]: Since 2.9 hasn't released yet :)

Comment (saratvemulapalli) [reply README.md:10]: It will be soon. 
Why not just add this to our version increment bot?

Comment (kotwanikunal) [reply README.md:10]: The increment bot does a bunch of file specific regex magic. We will have to write some good additions to the script for it. 
I cut a new issue here: https://github.com/opensearch-project/OpenSearch/issues/8851

Comment (kotwanikunal) [reply README.md]: Updated to 2.10

Comment (saratvemulapalli) [reply README.md]: Thanks, this should be fairly simple for the bot.

**Review by saratvemulapalli — APPROVED** [review]

**Pull request merged**